### DISCUSSION TITLE
Question about symlinking directories for include paths.

### DISCUSSION THREAD

**Discussion by mttbx** [SCons/scons#3891] [Q&A]
I use `SConscript('a', variant_dir='build/a')` to build the system. In `a/SConscript` how can I get the original dir 'a'? If I use `os.getcwd()` in `a/SConscript` I will get `build/a`.

**Comment by bdbaddog** [comment]
How do you want to use this info?
Depending on the answer there are different methods.

Also, generally it's not advised to have the variant dir be underneath the source dir.

so 
`SConscript('src/SConscript',variant_dir='build/a')
`

**Reply by mttbx** [reply]
Thank you for quick response. I want to do this thing because I want to create some symlinks dynamicly based on some options setted in parent directory.

**Reply by bdbaddog** [reply]
Symlinks to files? 
Which are sources, or targets of builders?

**Reply by bdbaddog** [reply]
Also.. if you're looking for help with SCons the best place (as indicated in the github issue template) is the users mailing list, the IRC channel, or the discord serve (https://discord.gg/bXVpWAy)

**Reply by mttbx** [reply]
Got it. The dir of src ('a') is kind of a module, and it is used in many project. However different project may have different configurations. So for example here's the structure in a:

a/dir1, a/dir2, a/dir, a/src.c

The symlink a/dir is pointed to a/dir1 or a/dir2 based on which configuration I used in the project. a/src.c depend on a/dir.

**Reply by bdbaddog** [reply]
Why not just use the correct dir name, instead of messing around with symlinks?

**Reply by mttbx** [reply]
Because a/src.c can be wrote more easier to use just a/dir/* instead of a/dir1/* or a/dir2/*. The build system in linux or uboot use this kind of strategy.

**Reply by mttbx** [reply]
If I don't use symlinks, I would need a lot of macros to decide which file I need to include.

**Reply by bdbaddog** [reply]
so in summary you want to make a symlink as follows:

`ln -s a/dir2 a/dir`
or
`ls -s a/dir1 a/dir`

**Reply by mttbx** [reply]
Yes, that's what I want to do.

**Reply by bdbaddog** [reply]
You wouldn't need macros... You just change the include path. (Via CPPPATH)..

**Reply by bdbaddog** [reply]
If this is just about including files, then honestly changing the CPPPATH is a MUCH better way to do this.
(And a very common practice)

**Reply by bdbaddog** [reply]
Are you building more than one variant at a time?

**Reply by bdbaddog** [reply]
Please come over the the discord server, it's much bettter for this.

**Reply by mttbx** [reply]
Well this can achieve it too. But is it possible to make symlinks? I prefer this way.

**Reply by mttbx** [reply]
In my contry, this site is too slow to be open...

**Reply by bdbaddog** [reply]
Just use CPPPATH... you'll thank me later.

Otherwise, I'd use Execute() in your SConstruct where you know which dir# you want, but remember you also need to rm the symlink before you reset it. Good luck.
I'm done

**Reply by bdbaddog** [reply]
See the manpage and/or the users guide.
Symlinks are NOT first class file types, so you'll have to run commands to create and remove them, and in general I advise against using them this way.

**Reply by mttbx** [reply]
Thank you for your advice!

**Reply by bdbaddog** [reply]
If the website is too slow, then join the users mailing and send your questions there.
There's not a ton of traffic and there are many people there to help.

**Reply by mttbx** [reply]
Got it, thank you.

**Comment by bdbaddog** [comment]
Closing
